packages/block-editor/src/components/link-control/index.js

```diff
@@ -196,7 +196,7 @@ function LinkControl( {
 	] = useInternalValue( value );
 
 	const valueHasChanges =
-		value && ! isShallowEqualObjects( internalControlValue, value );
+		value && ! isShallowEqual( internalControlValue, value );
```

Comment:
If this was to introduce a bug then the Link Control tests would fail 🤞 

It looks good anyway. Thanks for the ping 👍